martin-mbtiles/src/mbtiles.rs

```diff
@@ -378,32 +388,102 @@ impl Mbtiles {
         Err(MbtError::NoUniquenessConstraint(self.filepath.clone()))
     }
 
-    pub async fn validate_mbtiles<T>(&self, conn: &mut T) -> MbtResult<()>
+    pub async fn generate_global_hash<T>(&self, conn: &mut T) -> MbtResult<()>
     where
         for<'e> &'e mut T: SqliteExecutor<'e>,
     {
         let mbttype = self.detect_type(&mut *conn).await?;
 
-        let sql = match mbttype {
-            MbtType::Flat => {
-                return Err(IncorrectDataFormat(
-                    self.filepath().to_string(),
-                    &[MbtType::FlatWithHash, MbtType::Normalized],
-                    MbtType::Flat,
-                ));
-            }
-            MbtType::FlatWithHash => {
-                "SELECT * FROM tiles_with_hash WHERE tile_hash!=hex(md5(tile_data)) LIMIT 1;"
+        let select_from = match mbttype {
+            Flat => {
+                println!("Cannot generate global hash, no hash column in flat table format. Skipping global_hash generation...");
+                return Ok(());
             }
-            MbtType::Normalized => {
-                "SELECT * FROM images WHERE tile_id!=hex(md5(tile_data)) LIMIT 1;"
+            MbtType::FlatWithHash => "SELECT 'global_hash', hex(md5_concat(tile_hash)) FROM tiles_with_hash ORDER BY zoom_level, tile_column, tile_row;",
+            MbtType::Normalized => "SELECT 'global_hash', hex(md5_concat(images.tile_id)) FROM images JOIN map ON images.tile_id=map.tile_id ORDER BY zoom_level, tile_column, tile_row;"
```

Comment:
I am not sure you need to join two tables here. A quicker query `SELECT 'global_hash', hex(md5_concat(tile_id)) FROM images ORDER BY zoom_level, tile_column, tile_row;` would give you the same result. That said, you may want to validate that all foreign keys (`tile_id`) are valid - exist in both the images and map tables. And this could be part of the FULL validation. But it should be available even if the global hash is not requested.

For now, I think its better to do the simpler hash (without join), but we should add a full normalized table validation to check if there are `tile_id` in `map` that do not exist in `images`, or if there are any `tile_id` in images that are not being used by `map`.  TBH, I am not certain if this should be done with two querie, or if some sort of an outer join query could do this efficiently?

Reply:
I think as of right now, I will just do the simpler hash with no join and that is all